pkg/scaffold/resource/resource.go

```diff
@@ -75,7 +79,7 @@ func (r *Resource) Validate() error {
 	r.GroupImportSafe = strings.Replace(r.Group, "-", "", -1)
 	r.GroupImportSafe = strings.Replace(r.GroupImportSafe, ".", "", -1)
 
-	versionMatch := regexp.MustCompile("^v\\d+(alpha\\d+|beta\\d+)?$")
+	versionMatch := regexp.MustCompile(versionRegex)
```

Comment:
As you are already moving the pattern to a constant, why don't you move the regex to a `var` so that we do not compile it every time `Validate` gets called?

Adding after the `const` declaration
```
var (
	versionMatch := regexp.MustCompile(versionRegex)
)
```

Reply:
The var is used only this function, the has no need and should not be global.

Reply:
Indeed I moved the const for the above line.

Reply:
Using globals is not a problem as they are not exported. Compiling a regex is resource hungry and shouldn't be done multiple times unless the pattern is dynamic.